render-test/runner.cpp

```diff
@@ -383,10 +404,83 @@ bool TestRunner::checkRenderTestResults(mbgl::PremultipliedImage&& actualImage,
             return false;
         }
     }
+    // Check gfx metrics
+    for (const auto& expected : metadata.expectedMetrics.gfx) {
+        auto actual = metadata.metrics.gfx.find(expected.first);
+        if (actual == metadata.metrics.gfx.end()) {
+            metadata.errorMessage = "Failed to find gfx probe: " + expected.first;
+            return false;
+        }
+
+        auto check = [&metadata](
+                         const char* msg, const std::string& name, const auto& expected, const auto& actual) -> bool {
+            if (expected != actual) {
+                char msgBuf[512];
+                snprintf(msgBuf, sizeof(msgBuf), msg, name.c_str(), actual, expected);
```

Comment:
could `std::stringstream`  be used instead?

Reply:
I think the c-style printf formatting makes the code more readable (in this case) for the caller.